presto-main/src/main/java/com/facebook/presto/sql/planner/NodePartitioningManager.java

```diff
@@ -169,4 +172,35 @@ public NodePartitionMap getNodePartitioningMap(Session session, PartitioningHand
             return bucket;
         });
     }
+
+    private ConnectorBucketNodeMap getConnectorBucketNodeMap(Session session, PartitioningHandle partitioningHandle)
+    {
+        checkArgument(!(partitioningHandle.getConnectorHandle() instanceof SystemPartitioningHandle));
+
+        ConnectorNodePartitioningProvider partitioningProvider = partitioningProviders.get(partitioningHandle.getConnectorId().get());
+        checkArgument(partitioningProvider != null, "No partitioning provider for connector %s", partitioningHandle.getConnectorId().get());
+
+        ConnectorBucketNodeMap connectorBucketNodeMap = partitioningProvider.getBucketNodeMap(
+                partitioningHandle.getTransactionHandle().orElse(null),
+                session.toConnectorSession(),
+                partitioningHandle.getConnectorHandle());
+
+        checkArgument(connectorBucketNodeMap != null, "No partition map %s", partitioningHandle);
+        if (connectorBucketNodeMap.hasFixedMapping()) {
+            checkArgument(!connectorBucketNodeMap.getFixedMapping().isEmpty(), "Partition map is empty %s", partitioningHandle);
+        }
```

Comment:
The contract of `ConnectorBucketNodeMap` guarantees that this can't happen. I would remove this assertion. Optionally, you can add javadoc to `getBucketCount` and `getBucketCount` if you want to make the guarantee more obvious.

Reply:
Agreed. Now it's guarded by `ConnectorBucketNodeMap` constructor.